scipy/special/_logsumexp.py

```diff
@@ -93,43 +96,49 @@ def logsumexp(a, axis=None, b=None, keepdims=False, return_sign=False):
     1.6094379124341005, 1.6094379124341005
 
     """
-    a = _asarray_validated(a, check_finite=False)
-    if b is not None:
-        a, b = np.broadcast_arrays(a, b)
-        if np.any(b == 0):
-            a = a + 0.  # promote to at least float
-            a[b == 0] = -np.inf
+    xp = array_namespace(a, b)
+    a, b = xp_broadcast_promote(a, b, ensure_writeable=True)
+    axis = tuple(range(a.ndim)) if axis is None else axis
+
+    if b is not None and xp.any(b==0):
```

Comment:
How important is it to support staged-out computation as in JAX? Conditioning control flow on array contents (e.g. `if xp.any(b==0)`) will fail for staged-out operations.

Reply:
```suggestion
    if b is not None:
```
should be fine. 
That's OK with me.